src/main.cpp

```diff
@@ -13,55 +13,219 @@
+
+        // Set up process information
+        STARTUPINFOW si = { sizeof(si) };
+        PROCESS_INFORMATION pi;
+
+        // Create the process
+        if (CreateProcessW(NULL, &cmdLine[0], NULL, NULL, FALSE, 0, NULL, NULL, &si, &pi))
+        {
+            // Wait for the process to finish
+            WaitForSingleObject(pi.hProcess, INFINITE);
+
+            // Get the exit code
+            DWORD exitCode;
+            GetExitCodeProcess(pi.hProcess, &exitCode);
+
+            // Close process and thread handles
+            CloseHandle(pi.hProcess);
+            CloseHandle(pi.hThread);
+
+            if (exitCode == 0) {
+                fs::path converted_mdl = rmdlconv_out_dir / (mdl_path.filename().string() + "_new");
+                if (fs::exists(converted_mdl)) {
+                    // Remove original files
+                    for (const auto& ext : extensions) {
+                        fs::path to_remove = dst_base_path;
+                        to_remove.replace_extension(ext);
+                        fs::remove(to_remove);
+                    }
+                    // Move converted file
+                    fs::rename(converted_mdl, mdl_path);
+                    printf("Converted and replaced: %s\n", mdl_path.string().c_str());
+                }
+            }
+        }
+        else {
+            printf("Failed to start rmdlconv process. Error code: %d\n", GetLastError());
+        }
+
+        // Clean up rmdlconv_out directory
+        fs::remove_all(rmdlconv_out_dir);
+    }
 }
 
 
 int main(int argc, char* argv[]) {
-    if (argc != 3) {
+    if (argc != 4) {
         print_usage(argv[0]);
         return 0;
     }
-    char* in_filename  = argv[1];
-    char* out_filename = argv[2];
+    char* game_dir = argv[1];
+    char* in_filename = argv[2];
+    char* output_folder_with_extension = argv[3];
+
+    // Remove .bsp extension from in_filename if present
+    std::string in_filename_without_ext(in_filename);
+    if (in_filename_without_ext.length() > 4 && in_filename_without_ext.substr(in_filename_without_ext.length() - 4) == ".bsp") {
+        in_filename_without_ext = in_filename_without_ext.substr(0, in_filename_without_ext.length() - 4);
+    }
+    // Ditto
+    std::string output_folder(output_folder_with_extension);
+    if (output_folder.length() > 4 && output_folder.substr(output_folder.length() - 4) == ".bsp") {
+        output_folder = output_folder.substr(0, output_folder.length() - 4);
+    }
+    // Initialize the file system with the provided game directory
+    if (!InitFileSystem(game_dir, in_filename)) {
+        fprintf(stderr, "Failed to initialize filesystem\n");
+        return 1;
+    }
+
+    // Create output directories
+    fs::path maps_output_dir = fs::path(output_folder) / "maps";
+    try {
+        fs::create_directories(maps_output_dir);
+    }
+    catch (const fs::filesystem_error& e) {
+        std::cerr << "Failed to create output directory: " << e.what() << std::endl;
```

Comment:
I'd prefer to use `fprintf` over `<<`

Reply:
keeps it consistent with the rest of the code
also means any variable -> text conversion is explicit